workflow/controller/operator.go

```diff
@@ -164,6 +167,16 @@ func (woc *wfOperationCtx) operate() {
 
 	woc.setGlobalParameters()
 
+	if woc.wf.Spec.ArtifactRepositoryRef != nil {
+		repoReference := woc.wf.Spec.ArtifactRepositoryRef
+		repo, err := getArtifactRepositoryRef(woc.controller, repoReference.ConfigMap, repoReference.Key)
+		if err == nil {
+			woc.artifactRepository = repo
+		} else {
+			woc.log.Errorf("Failed to load artifact repository configMap: %+v", err)
```

Comment:
I think we should fail the workflow if we can't dereference the repository.

Reply:
Done.